Subscribers/Article.php

```diff
@@ -101,10 +102,60 @@ public static function getSubscribedEvents()
             'Enlight_Controller_Action_PostDispatch_Backend_Article' => 'extendBackendArticle',
             'Enlight_Controller_Action_PreDispatch_Backend_Article' => 'preBackendArticle',
             'Enlight_Controller_Action_PostDispatch_Frontend_Detail' => 'modifyConnectArticle',
-            'Enlight_Controller_Action_PreDispatch_Frontend_Detail' => 'extendFrontendArticle'
+            'Enlight_Controller_Action_PreDispatch_Frontend_Detail' => 'extendFrontendArticle',
+            'Shopware_Modules_Basket_AddArticle_Start' => 'checkSupplierPluginAvailability'
         ];
     }
 
+    /**
+     * @param \Enlight_Event_EventArgs $args
+     * @return bool
+     * @throws \Exception
+     */
+    public function checkSupplierPluginAvailability(\Enlight_Event_EventArgs $args)
+    {
+        $articleDetail = $this->helper->getDetailByNumber($args->getId());
+        $articleDetailId = $articleDetail->getId();
+
+        if ($this->helper->isRemoteArticleDetail($articleDetailId)) {
+            $shopProductId = $this->helper->getShopProductId($articleDetailId);
+            $shopId = $shopProductId->shopId;
+
+            return $this->pingRemoteShop($shopId);
+        };
+    }
+
+    /**
+     * @param $shopId
+     * @return bool
+     * @throws \Exception
+     */
+    public function pingRemoteShop($shopId)
+    {
+        /**
+         * @var RemoteShopService $remoteShopService
+         * @todo: refactor when using 5.2 plugin base.
+         */
+        $remoteShopService = Shopware()->Container()->get('swagconnect.remote_shop_service');
+
+        if (!$remoteShopService->isPingRemoteShopSuccessful($shopId)) {
+            $this->createBasketInfoMessagesOnFailingRemoteShopPing();
+            return false;
```

Comment:
remove too?

Reply:
This is needed to prevent the article being added to the basket (Shopware_Modules_Basket_AddArticle_Start event). Without return the product would be added anyway.